### PR TITLE
Expand ABI/API explanation in HACKING

### LINKED ISSUE #1475: “Before the 1.0 release series” in HACKING
The [HACKING](https://github.com/geany/geany/blob/master/HACKING) file has this:

> Before the 1.0 release series, the ABI can change when necessary, and
even the API can change.

Obviously Geany has long switched to 1.x versions. Does this text still apply?

### PR DESCRIPTION
**Pull request by xiota**
Integrate @elextr ['s explanation](https://github.com/geany/geany/issues/1475#issuecomment-296629551) of ABI/API into the `HACKING` document.  Resolves #1475.

### CODE REVIEW

**Comment by eht16** [comment]
LGTM.

File: HACKING [hunk 1]
```diff
@@ -126,18 +126,26 @@ the existing elements stay in place - this will keep the ABI stable.
...
+plugins that need a lower number from working.
 
 If you're reordering or changing existing elements of structs that are
 used as part of the plugin API, you must increment GEANY_ABI_VERSION
 in plugindata.h. This is usually not needed if you're just appending
 fields to structs. The GEANY_API_VERSION value should be incremented
 for any changes to the plugin API, including appending elements.
 
-If you're in any doubt when making changes to plugin API code, just ask us.
+If you're in doubt when making changes to plugin API code, just ask us.
```
Comment (elextr) [inline HACKING:148]: Not sure why you took the "any" out?

**Comment by elextr** [comment]
Other than the query LGTM.

File: HACKING (hunk 1 above)
Comment (xiota) [reply HACKING:148]: Took it out for line length.  Any is implied.  Can put it back if you want.

Comment (elextr) [reply HACKING:148]: The "any" just adds emphasis, we prefer people ask than break the API, so prefer it stays.

Comment (xiota) [reply HACKING]: okay.

**Pull request merged**